gremlin-server/src/main/java/org/apache/tinkerpop/gremlin/server/GraphManager.java

```diff
@@ -21,139 +21,87 @@
      */
-    public void rollbackAll() {
-        graphs.entrySet().forEach(e -> {
-            final Graph graph = e.getValue();
-            if (graph.features().graph().supportsTransactions() && graph.tx().isOpen())
-                graph.tx().rollback();
-        });
-    }
+    public void rollbackAll();
 
     /**
      * Selectively rollback transactions on the specified graphs or the graphs of traversal sources.
      */
-    public void rollback(final Set<String> graphSourceNamesToCloseTxOn) {
-        closeTx(graphSourceNamesToCloseTxOn, Transaction.Status.ROLLBACK);
-    }
+    public void rollback(final Set<String> graphSourceNamesToCloseTxOn);
 
     /**
      * Commit transactions across all {@link Graph} objects.
      */
-    public void commitAll() {
-        graphs.entrySet().forEach(e -> {
-            final Graph graph = e.getValue();
-            if (graph.features().graph().supportsTransactions() && graph.tx().isOpen())
-                graph.tx().commit();
-        });
-    }
+    public void commitAll();
 
     /**
      * Selectively commit transactions on the specified graphs or the graphs of traversal sources.
      */
-    public void commit(final Set<String> graphSourceNamesToCloseTxOn) {
-        closeTx(graphSourceNamesToCloseTxOn, Transaction.Status.COMMIT);
-    }
+    public void commit(final Set<String> graphSourceNamesToCloseTxOn);
 
     /**
-     * Selectively close transactions on the specified graphs or the graphs of traversal sources.
+     * Implementation that allows for custom graph-opening implementations.
      */
-    private void closeTx(final Set<String> graphSourceNamesToCloseTxOn, final Transaction.Status tx) {
-        final Set<Graph> graphsToCloseTxOn = new HashSet<>();
-
-        // by the time this method has been called, it should be validated that the source/graph is present.
-        // might be possible that it could have been removed dynamically, but that i'm not sure how one would do
-        // that as of right now unless they were embedded in which case they'd need to know what they were doing
-        // anyway
-        graphSourceNamesToCloseTxOn.forEach(r -> {
-            if (graphs.containsKey(r))
-                graphsToCloseTxOn.add(graphs.get(r));
-            else
-                graphsToCloseTxOn.add(traversalSources.get(r).getGraph());
-        });
+    public Graph openGraph(String graphName, Supplier<Graph> supplier);
 
-        graphsToCloseTxOn.forEach(graph -> {
-            if (graph.features().graph().supportsTransactions() && graph.tx().isOpen()) {
-                if (tx == Transaction.Status.COMMIT)
-                    graph.tx().commit();
-                else
-                    graph.tx().rollback();
-            }
-        });
-    }
-}
\ No newline at end of file
+    /**
+     * Implementation that allows for custom graph-closing implementations.
+     */
+    public void closeGraph(Graph graph);
```

Comment:
oh, i see what you're saying. i think that if you add both, you should make a default implementation on the one with the String argument to use `getGraph()` and then pass that to the `closeGraph(Graph graph)` method.

was there a reason you went with `closeGraph()` as opposed to `removeGraph()`. it seems like the `GraphManager` is about managing `Graph` instances that it holds and "remove" is a better term for the behavior `GraphManager` is supposed to be doing there.  @pluradj i think you'd suggested this addition - perhaps you could chime in on this.